### PR TITLE
Add bll for UserProfileService #4

### PR DESCRIPTION
**Pull request by vladzhorov**

### CODE REVIEW

File: ServiceFinder/ServiceFinder.BLL/Abstarctions/Services/IGenericService.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿namespace ServiceFinder.BLL.Abstractions.Services
+{
+    public interface IGenericService<TEntity, TModel>
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.BLL/Abstarctions/Services/IGenericService.cs:3]: 'TEntity' is not used in the interface.

File: ServiceFinder/ServiceFinder.BLL/Models/Review.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+{
+    public class Review
+    {
+        public Guid Id { get; set; }
+        public Guid ServiceId { get; set; }
+        public float Rating { get; set; }
+        public string? Comment { get; set; }
+        public DateTime CreatedAt { get; set; }
+        public DateTime UpdatedAt { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.BLL/Models/Review.cs:11]: Do you need this 'IsDeleted' flag here?

File: ServiceFinder/ServiceFinder.BLL/Models/ServiceCategory.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace ServiceFinder.BLL.Models
+{
+    public class ServiceCategory
+    {
+        public Guid Id { get; set; }
+        public string? Name { get; set; }
+        public string? Description { get; set; }
+        public DateTime CreatedAt { get; set; }
+        public DateTime UpdatedAt { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.BLL/Models/ServiceCategory.cs:10]: same

File: ServiceFinder/ServiceFinder.BLL/Services/GenericService.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+using ServiceFinder.DAL.Interfaces;
+
+namespace ServiceFinder.BLL.Services
+{
+    public class GenericService<TEntity, TModel> : IGenericService<TEntity, TModel> where TEntity : class
+    {
+        protected readonly IRepository<TEntity> _repository;
+        protected readonly IMapper _mapper;
+        public GenericService(IRepository<TEntity> repository, IMapper mapper)
+        {
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.BLL/Services/GenericService.cs:12]: formatting

File: ServiceFinder/ServiceFinder.BLL/Services/GenericService.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            _repository = repository;
+            _mapper = mapper;
+        }
+
+        public async virtual Task<TModel> GetByIdAsync(Guid id, CancellationToken cancellationToken)
+        {
+            var entity = await _repository.GetByIdAsync(id, CancellationToken.None);
+            var model = _mapper.Map<TModel>(entity);
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.BLL/Services/GenericService.cs:20]: what's the purpose of the cancellation token if you do not use it in repo, lol?

File: ServiceFinder/ServiceFinder.BLL/Services/GenericService.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            var entity = _mapper.Map<TEntity>(model);
+            var result = await _repository.UpdateAsync(entity, cancellationToken);
+            return _mapper.Map<TModel>(result);
+        }
+
+        public async virtual Task DeleteAsync(Guid id, CancellationToken cancellationToken)
+        {
+            var entity = await _repository.GetByIdAsync(id, cancellationToken);
+            await _repository.DeleteAsync(entity, cancellationToken);
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.BLL/Services/GenericService.cs:41]: fix warnings

File: ServiceFinder/ServiceFinder.BLL/Models/UserProfile.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+{
+    public class UserProfile
+    {
+        public Guid Id { get; set; }
+        public string? PhotoURL { get; set; }
+        public string? PhoneNumber { get; set; }
+        public float Rating { get; set; }
+        public DateTime CreatedAt { get; set; }
+        public DateTime UpdatedAt { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.BLL/Models/UserProfile.cs:11]: same

File: ServiceFinder/ServiceFinder.BLL/Models/Service.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        public ServiceCategory? Category { get; set; }
+        public string? Description { get; set; }
+        public decimal Price { get; set; }
+        public TimeSpan Duration { get; set; }
+        public float Rating { get; set; }
+        public string? Location { get; set; }
+        public ICollection<Review>? Reviews { get; set; }
+        public DateTime CreatedAt { get; set; }
+        public DateTime UpdatedAt { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.BLL/Models/Service.cs:18]: same

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by moltenessence — APPROVED** [review]

**Review by pavelGagarin0870 — APPROVED** [review]

**Pull request merged**